crates/core/src/storage/mod.rs

```diff
@@ -168,7 +198,28 @@ mod tests {
         let store = InMemory::new();
         let path = Path::parse("/databases/foo/bar").expect("Failed to parse path");
 
-        let prefixed = url_prefix_handler(store, path);
-        assert!(prefixed.is_ok());
+        let prefixed = url_prefix_handler(store, path.clone());
+
+        assert_eq!(
+            String::from("PrefixObjectStore(databases/foo/bar)"),
+            format!("{prefixed}")
+        );
+    }
+
+    #[test]
+    fn test_limit_store_handler() {
+        let store = InMemory::new();
+
+        let options = StorageOptions(HashMap::from_iter(vec![(
+            "OBJECT_STORE_CONCURRENCY_LIMIT".into(),
+            "500".into(),
+        )]));
+
+        let limited = limit_store_handler(store, &options);
+
+        assert_eq!(
+            String::from("LimitStore(500, InMemory)"),
+            format!("{limited}")
```

Comment:
I used the fact that each ObjectStore has a Display trait implementation, I didn't find a better way to test this